lib/travis/build/script/c.rb

```diff
@@ -43,7 +43,13 @@ def use_directory_cache?
         private
 
           def compiler
-            config[:compiler].to_s
+            case config[:compiler].to_s
+            when /^gcc/i, /^clang/i
+              config[:compiler].to_s
+            else
+              { 'linux' => 'gcc', 
```

Comment:
This construct is rather unsightly.

A more straightforward

```ruby
case config[:os]
when 'osx'
  'clang'
when 'linux'
  'gcc'
end
```

is preferred.